[Docs] Prepare docs for refactoring (#678)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/install.md

## Set up access to a large language model

> Vizro-AI supports **any** model that is available via [Langchain's `BaseChatModel` class](https://api.python.langchain.com/en/latest/language_models/langchain_core.language_models.chat_models.BaseChatModel.html#langchain_core.language_models.chat_models.BaseChatModel), and that has the `with_structured_output` method implemented. An overview of the [most common vendor models supporting this functionality](https://python.langchain.com/v0.2/docs/integrations/chat/) can be found in Langchain's documentation.

Author: I think not possible 😬 , but good spot!

> Vizro-AI supports **any** model that is available via [Langchain's `BaseChatModel` class](https://api.python.langchain.com/en/latest/language_models/langchain_core.language_models.chat_models.BaseChatModel.html#langchain_core.language_models.chat_models.BaseChatModel), and that has the `with_structured_output` method implemented. An overview of the [most common vendor models supporting this functionality](https://python.langchain.com/v0.2/docs/integrations/chat/) can be found in Langchain's documentation.

Author: All those models may be subclassed from `BaseChatModel`, but importantly not all have `with_structured_output` implemented. Is that confusing from my text?

Good link, I will include that one as well!

In general I think we should when time permits dig a bit deeper into that method and check:
- what are the failure modes (I realise there can be a `None` return, which we do not account for
- is langchain leveraging the newest features around this capability, e.g. OpenAIs guaranteed schema adherence

The documentation provides step-by-step instructions for setting up the API key as an environment
variable, on operating systems including Windows and MacOS.

!!!note

>     Sometimes setting up the `.env` file can be fiddly. If there is continued struggle, remember that one can always provide the API key directly to the instantiated model. See [our user guide](./customize-vizro-ai.md#setting-model-via-class-for-additional-configuration) for this option. Remember to still not commit this API key to any public space!

Reviewer: ```suggestion
    Sometimes setting up the `.env` file can be fiddly. If necessary, you can provide the API key directly to the instantiated model. See [our user guide](./customize-vizro-ai.md#setting-model-via-class-for-additional-configuration) for this option. Remember not to commit this API key to any public space!
```
